wallet/src/account.rs

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::key_chain::{AccountKeyChain, KeyPurpose};
+use crate::{WalletError, WalletResult};
+use common::address::Address;
+use common::chain::{ChainConfig, OutPoint, Transaction, TxOutput};
+use common::primitives::{Id, Idable};
+use std::collections::BTreeMap;
+use std::sync::Arc;
+use storage::Backend;
+use utxo::Utxo;
+use wallet_storage::{StoreTxRo, StoreTxRw, WalletStorageRead, WalletStorageWrite};
+use wallet_types::{AccountId, AccountOutPointId, AccountTxId, TxState, WalletTx};
+
+#[allow(dead_code)] // TODO remove
```

Comment:
If only chain_config is dead_code, you can put it above chain_config only. No need to deem the whole struct as dead_code.